framework/jobs/src/main/java/org/apache/cloudstack/framework/jobs/impl/AsyncJobManagerImpl.java

```diff
@@ -1049,8 +1054,8 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
                             }
                         }
                     }
-                    List<SnapshotDetailsVO> snapshotList = _snapshotDetailsDao.findDetails(AsyncJob.Constants.MS_ID, Long.toString(msid), false);
-                    for (SnapshotDetailsVO snapshotDetailsVO : snapshotList) {
+                    final List<SnapshotDetailsVO> snapshotList = _snapshotDetailsDao.findDetails(AsyncJob.Constants.MS_ID, Long.toString(msid), false);
```

Comment:
I think it's a good defensive programming style to keep assignment restrictions unless needed.

Reply:
Sorry if I sound blunt, but yes I know this "excuse". My question is, what are the cases that the use of these final keywords is helping us to avoid?

Is it the case that we have these monstrous methods (with 100+, or 500+ lines)? And then you are trying to help programmers to identify when he/she is changing a variable that was previously assigned somewhere else in the middle of a big chunk of code. 

If that is the only reason for us to use this keyword, I do prefer other options such as using shorter and concise methods, which can be unit tested. Then, we can catch problems such as wrong assignments. Otherwise, we are just adding code that is not needed.

Reply:
Best practice is to make objects immutable and it makes the code safer and more predictable.

Reply:
That is exactly my point. I have seen this indiscriminate use of the `final` keyword in some other places as if it would greatly improve our code base. This is also not the first time I call attention regarding the use of final.

I really do not understand when I see people slapping `final` keywords in a method that has 100+ lines that can be extracted into smaller blocks. Then, we hear that it is a defensive programming style or some sort of best practice. I know, those expressions sound nice and shiny, but simply adding the `final` keyword makes no improvement what so ever. A poorly coded method will remain poorly coded with or without `final` keywords in its variables.